### PR TITLE
Moves custom Vite config in project root dir

### PR DESCRIPTION
**Pull request by infomiho**
In the new structure, the natural place for `vite.config.{ts,js}` is in the project root dir since that's where all other JS projects keep it.

I've adjusted the code to find use the custom Vite config if the user provides it in the project root dir. I had to put a `@ts-ignore` on top of the import since we are importing a file outside of the `rootDir` of the web app.

### CODE REVIEW

File: waspc/src/Wasp/Generator/WebAppGenerator.hs [hunk 1]
```diff
@@ -292,10 +292,10 @@ genEnvValidationScript =
 genWebSockets :: AppSpec -> Generator [FileDraft]
 genWebSockets spec
   | AS.WS.areWebSocketsUsed spec =
-    sequence
-      [ genFileCopy [relfile|webSocket.ts|],
-        genWebSocketProvider spec
-      ]
+      sequence
```
Comment (infomiho) [inline waspc/src/Wasp/Generator/WebAppGenerator.hs:295]: Ormolu formatted this for me. I believe that's something that Ormolu complained about in the CI anyways.

File: waspc/examples/todo-typescript/vite.config.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { defineConfig } from "vite";
```
Comment (infomiho) [inline waspc/examples/todo-typescript/vite.config.ts:1]: Custom Vite config in the prototype root.

File: waspc/src/Wasp/Generator/WebAppGenerator.hs [hunk 3]
```diff
@@ -320,10 +320,13 @@ genViteConfig spec = return $ C.mkTmplFdWithData tmplFile tmplData
           "defaultClientPort" .= C.defaultClientPort
         ]
 
-    makeCustomViteConfigJsImport :: Path' (Rel SourceExternalCodeDir) File' -> JsImport
+    makeCustomViteConfigJsImport :: Path' (Rel WaspProjectDir) File' -> JsImport
```
Comment (infomiho) [inline waspc/src/Wasp/Generator/WebAppGenerator.hs:323]: This function creates an import for the custom config, it does it relatively from the `web-app` to the user project dir.

File: waspc/src/Wasp/Generator/WebAppGenerator.hs [hunk 4]
```diff
@@ -320,10 +320,13 @@ genViteConfig spec = return $ C.mkTmplFdWithData tmplFile tmplData
...
     makeCustomViteConfigJsImport pathToConfig = makeJsImport (RelativeImportPath importPath) importName
       where
-        importPath = C.toViteImportPath $ fromJust $ SP.relFileToPosix pathToConfigInSrc
-        pathToConfigInSrc = SP.castRel $ C.webAppSrcDirInWebAppRootDir </> SP.castRel pathToConfig
+        importPath = SP.castRel $ C.toViteImportPath relPathToConfigInProjectDir
+        relPathToConfigInProjectDir = relPathFromWebAppRootDirWaspProjectDir </> (fromJust . SP.relFileToPosix $ pathToConfig)
+
+        relPathFromWebAppRootDirWaspProjectDir :: Path Posix (Rel C.WebAppRootDir) (Dir WaspProjectDir)
+        relPathFromWebAppRootDirWaspProjectDir = [reldirP|../../..|]
```
Comment (sodic) [inline waspc/src/Wasp/Generator/WebAppGenerator.hs:328]: We already have this knowledge in our codebase. You can get what you need by combining `webAppRootDirInProjectRootDir` (in WebAppGenerator.Common) and `webAppRootDirInProjectRootDir` (in Project.Common).

Functions you'll need:
- `relDirToPosix`: https://hackage.haskell.org/package/strong-path-1.1.4.0/docs/StrongPath.html#g:26
- `reversePosixPath` in `FilePath.Extra`

Comment (infomiho) [reply waspc/src/Wasp/Generator/WebAppGenerator.hs:328]: Nice stuff, I'll do it like this 👍

**Review by sodic — APPROVED** [review]
Looks good, you can merge it.

**Pull request merged**